### PR TITLE
Make DataInterceptorResolver#resolve safe for concurrent usage.

### PR DESCRIPTION
**Pull request by Hc747**
- DataInterceptorResolver: utilise Map#compute in DataInterceptorResolver#resolve to ensure that resolution of DataInterceptors is safe for concurrent use.

- DataInterceptorResolver: misc. polishing/refactoring.

### CODE REVIEW

**Comment by dstepanov** [comment]
Did you have some problems with it?

File: data-model/src/main/java/io/micronaut/data/intercept/DataInterceptorResolver.java [hunk 1]
```diff
@@ -49,76 +49,72 @@
...
-             return dataInterceptor;
-         }
-         String dataSourceName = context.stringValue(Repository.class)
-             .orElseGet(() -> injectionPoint == null ? null : injectionPoint.getAnnotationMetadata().stringValue(Repository.class).orElse(null));
-         Class<?> operationsType = context.classValue(RepositoryConfiguration.class, "operations")
-             .orElse(PrimaryRepositoryOperations.class);
-         Class<?> interceptorType = context
-             .classValue(DataMethod.class, DataMethod.META_MEMBER_INTERCEPTOR)
-             .orElse(null);
+        return interceptors.compute(key, (k, interceptor) -> {
```
Comment (dstepanov) [inline data-model/src/main/java/io/micronaut/data/intercept/DataInterceptorResolver.java:62]: Use `computeIfAbsent`

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: data-model/src/main/java/io/micronaut/data/intercept/DataInterceptorResolver.java (hunk 1 above)
Comment (Hc747) [reply data-model/src/main/java/io/micronaut/data/intercept/DataInterceptorResolver.java:62]: Good catch!

**Comment by Hc747** [comment]
> Did you have some problems with it?

No, but I had been following #368 for a while, and had a look into your solution to it. 👍 
It's a minor improvement, but, using the compute method over a sequence of get/put operations ensures they are atomic and linearisable.

**Comment by dstepanov** [comment]
@Hc747 Thanks, sometimes we have separate `get/put` calls because `compute*` methods don't support recursive invocation but this it's not the case.

**Pull request merged**